token/client/src/token.rs

```diff
@@ -595,23 +595,24 @@ where
 
         instructions.extend_from_slice(token_instructions);
 
-        let blockhash = if let (Some(nonce_account), Some(nonce_authority), Some(nonce_blockhash)) = (
-            self.nonce_account,
-            &self.nonce_authority,
-            self.nonce_blockhash,
-        ) {
-            let nonce_instruction = system_instruction::advance_nonce_account(
-                &nonce_account,
-                &nonce_authority.pubkey(),
-            );
-            instructions.insert(0, nonce_instruction);
-            nonce_blockhash
-        } else {
-            self.client
-                .get_latest_blockhash()
-                .await
-                .map_err(TokenError::Client)?
-        };
+        let mut blockhash =
+            if let (Some(nonce_account), Some(nonce_authority), Some(nonce_blockhash)) = (
+                self.nonce_account,
+                &self.nonce_authority,
+                self.nonce_blockhash,
+            ) {
+                let nonce_instruction = system_instruction::advance_nonce_account(
+                    &nonce_account,
+                    &nonce_authority.pubkey(),
+                );
+                instructions.insert(0, nonce_instruction);
+                nonce_blockhash
+            } else {
+                self.client
+                    .get_latest_blockhash()
+                    .await
+                    .map_err(TokenError::Client)?
+            };
```

Comment:
The token-client is also used by the CLI, where we shouldn't be over-fetching blockhashes. Rather than fetch it twice, if you really think this is the issue, we should just move the whole blockhash fetch to later, and change this to 
```suggestion
            if let (Some(nonce_account), Some(nonce_authority), Some(nonce_blockhash)) = (
                self.nonce_account,
                &self.nonce_authority,
                self.nonce_blockhash,
            ) {
                let nonce_instruction = system_instruction::advance_nonce_account(
                    &nonce_account,
                    &nonce_authority.pubkey(),
                );
                instructions.insert(0, nonce_instruction);
            }
```

And then later on, get the hash all at once with my other suggestion. What do you think?